spec/tests/manual.fmf

```diff
@@ -1,50 +1,32 @@
-summary: Manual test execution details
+summary: Test automation state
 
 story:
-    As a tester I want to store detailed manual instructions
-    covering test scenario so that I can perform them manually
-    before the test can be fully automated.
+    As a tester I need to store detailed manual instructions covering
+    test scenario I have to perform manually.
 
 description: |
-    Sometimes test scenarios cannot be fully automated or it takes
-    time to fully automate them. For such cases the ``manual``
-    attribute can be used to describe test steps in detail. It
-    should be a ``dictionary`` with the following keys, each one
-    of them should be a ``string``.
-
-    setup
-        Environment preparation instructions.
-    step
-        Test step, action to be performed.
-    expect
-        Expected results of the test step.
-    cleanup
-        Environment cleanup instructions.
-
-    If there are several individual steps they can be stored, with
-    their corresponding expected results, as a list of dictionaries
-    under the ``steps`` key.
+    Attribute marks whether this test needs human interaction during its execution.
+
+    Such tests are not likely to be executed in automation pipelines.
+    In the future they can be executed in semi-automated way, waiting on human interaction.
+
+    It's value should be a ``bool``. The default value is ``false``.
+
+    Attribute :ref:`spec/tests/test` should point to a Markdown document in defined format.
+    Names and types of sections are exactly as bellow and reserved for TMT purposes.
+    Any other features of the Markdown are free to be used but TMT will just show them.
+
+    Reserved sections:
+      ``# Setup`` Optional heading under which any necessary preparation actions which are not part of the test itself are documented
+      ``# Test`` Required heading marks start of the test itself
+      ``### Step`` Required heading of level 3 marking single step of the test, should be in pair with Expect heading
+      ``### Expect`` Required heading of level 3 marking expected outcome of the previous step
+      ``# Cleanup`` Optional heading under which any cleanup actions which are not part of the test itself are documented
```

Comment:
One more option came to my mind: What about using only two levels for the basic structure? Only `Setup` and `Cleanup` would be strictly defined for level 1 heading and any other level 1 heading would be considered a test.

This would allow users to name test sections freely and save one additional level for writing, parsing and explaining. The resulting structure would look like this:

    # Setup
    Preparation instructions.

    # Custom Test Section
    ## Step
    Do this and that.
    ## Expect
    Check this and that.

    # Another Test Section
    ## Step
    Do this and that.
    ## Expect
    Check this and that.

    # Cleanup
    Cleanup instructions.

Minimal manual test example:

    # Test
    ## Step
    Do this and that.
    ## Expect
    Check this and that.

What do you think?

Reply:
ACK